modules/account-lib/test/unit/coin/near/transactionBuilder/stakingActivateBuilder.ts

```diff
@@ -0,0 +1,93 @@
+        .amount(amount)
+        .gas(gas)
+        .sender(testData.accounts.account1.address, testData.accounts.account1.publicKey)
+        .receiverId(validatorContractAddress)
+        .recentBlockHash(testData.blockHash.block1)
+        .nonce(1);
+      txBuilder.sign({ key: testData.accounts.account1.secretKey });
+      const tx = await txBuilder.build();
+      tx.inputs.length.should.equal(1);
+      tx.inputs[0].should.deepEqual({
+        address: testData.accounts.account1.address,
+        value: amount,
+        coin: 'tnear',
+      });
+      tx.outputs.length.should.equal(1);
+      tx.outputs[0].should.deepEqual({
+        address: validatorContractAddress,
+        value: amount,
+        coin: 'tnear',
+      });
+
+      should.equal(tx.type, TransactionType.StakingActivate);
+      const rawTx = tx.toBroadcastFormat();
+      should.equal(rawTx, testData.rawTx.stakingActivate.signed);
+    });
+
+    it('build a create and delegate staking unsigned tx', async () => {
+      const txBuilder = factory.getStakingActivateBuilder();
+      txBuilder
+        .amount(amount)
+        .gas(gas)
+        .sender(testData.accounts.account1.address, testData.accounts.account1.publicKey)
+        .receiverId(validatorContractAddress)
+        .recentBlockHash(testData.blockHash.block1)
+        .nonce(1);
+      const tx = await txBuilder.build();
+      tx.inputs.length.should.equal(1);
+      tx.inputs[0].should.deepEqual({
+        address: testData.accounts.account1.address,
+        value: amount,
+        coin: 'tnear',
+      });
+      tx.outputs.length.should.equal(1);
+      tx.outputs[0].should.deepEqual({
+        address: validatorContractAddress,
+        value: amount,
+        coin: 'tnear',
+      });
+
+      should.equal(tx.type, TransactionType.StakingActivate);
+      const rawTx = tx.toBroadcastFormat();
+      should.equal(rawTx, testData.rawTx.stakingActivate.unsigned);
+    });
+
+    it('should fail to create and delegate staking with missing gas', async () => {
+      const txBuilder = factory.getStakingActivateBuilder();
+      txBuilder
+        .amount(amount)
+        .sender(testData.accounts.account1.address, testData.accounts.account1.publicKey)
+        .receiverId(validatorContractAddress)
+        .recentBlockHash(testData.blockHash.block1)
+        .nonce(1);
+      await txBuilder.build().should.be.rejectedWith('gas is required before building staking activate');
+    });
+
+    it('should fail to create and delegate staking with missing amount', async () => {
+      const txBuilder = factory.getStakingActivateBuilder();
+      txBuilder
+        .gas(gas)
+        .sender(testData.accounts.account1.address, testData.accounts.account1.publicKey)
+        .receiverId(validatorContractAddress)
+        .recentBlockHash(testData.blockHash.block1)
+        .nonce(1);
+      await txBuilder.build().should.be.rejectedWith('amount is required before building staking activate');
+    });
+  });
+});
```

Comment:
You should also test fromImplementation for staking transactions.

Reply:
good catch!